protocol/cosi.go

```diff
@@ -213,37 +209,29 @@ func (pc *ProtocolCosi) handleCommitment(in *Commitment) error {
 		return pc.commitmentHook(pc.tempCommitment)
 	}
 
-	// or make continue the cosi protocol
-	commits := make([]*cosi.Commitment, len(pc.tempCommitment))
-	secretVar := pc.Suite().Point().Null()
-	for i := range pc.tempCommitment {
-		secretVar.Add(secretVar, pc.tempCommitment[i].Commitment.Commitment)
-		commits[i] = pc.tempCommitment[i].Commitment
-	}
-
 	// go to Commit()
-	out := pc.Cosi.Commit(nil, commits)
-	secretVar.Add(secretVar, pc.Cosi.GetCommitment())
+	out := pc.Cosi.Commit(nil, pc.tempCommitment)
+
 	// if we are the root, we need to start the Challenge
 	if pc.IsRoot() {
 		return pc.StartChallenge()
 	}
 
 	// otherwise send it to parent
 	outMsg := &Commitment{
-		Commitment: out,
+		Comm: out,
 	}
 	return pc.SendTo(pc.Parent(), outMsg)
 }
 
 // StartChallenge start the challenge phase. Typically called by the Root ;)
-func (pc *ProtocolCosi) StartChallenge() error {
+func (pc *CoSi) StartChallenge() error {
```

Comment:
I'm not sure about this one because for the moment `dedis/cosi/protocols` and `dedis/crypto/cosi` have *compatible* ~ *essentially the same* API: one function for starting a phase and one for relaying a phase. I don't think it would make much sense to change here but not in `dedis/crypto/cosi`. Moreover, I always prefer explicit over implicit, having these nil parameters is not a good idea from my PoV since it requires explicit attention when you are root or leaf as what you should do..
But again, let the majority decide. If you think that makes sense to merge the phases into one function each *here and in dedis/crypto/cosi*, then that makes two of you with Bryan so I would abide :)